### PR TITLE
Provide precompiled bundle

### PR DESCRIPTION
**Pull request by Nachtalb**

### CODE REVIEW

**Comment by Nachtalb** [comment]
Rebased to latest fix-test branch to get a working PR with no failing tests

File: ftw/sliderblock/browser/resources/sliderblock.js [hunk 1]
```diff
@@ -32,7 +32,7 @@
           var settings = sliderPane.parent().data("settings");
```
Comment (jone) [inline ftw/sliderblock/browser/resources/sliderblock.js:36]: Does this work in Plone 4 and 5? Shouldn't we use the pattern described in the docu: https://devdocs.4teamwork.ch/plone/javascript/ftw-resource-bundling-plone5/#transform-our-javascript-in-the-product ?
Not sure..

File: ftw/sliderblock/profiles/default_plone5/registry.xml [hunk 2]
```diff
@@ -25,21 +25,12 @@
...
              interface='Products.CMFPlone.interfaces.IBundleRegistry'>
-        <value key="resources">
-            <element>ftw-sliderblock-js</element>
-        </value>
         <value key="enabled">True</value>
         <value key="depends">plone-legacy</value>
         <value key="compile">False</value>
-        <value key="jscompilation">++plone++sliderblock/ftw-sliderblock-compiled.js</value>
+        <value key="jscompilation">++resource++ftw.sliderblock/sliderblock-compiled.js</value>
         <value key="merge_with">default</value>
```
Comment (jone) [inline ftw/sliderblock/profiles/default_plone5/registry.xml:34]: According to https://devdocs.4teamwork.ch/plone/javascript/ftw-resource-bundling-plone5/#declare-a-bundle-that-is-built-externally you need a `last_compilation` entry.

File: ftw/sliderblock/upgrades/20191212135955_provide_precompiled_bundle/registry.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<registry>
+
+    <records prefix="plone.bundles/ftw-sliderblock-resources"
+             interface='Products.CMFPlone.interfaces.IBundleRegistry'>
+        <value key="resources" purge="True"></value>
+        <value key="enabled">True</value>
+        <value key="depends">plone-legacy</value>
+        <value key="compile">False</value>
+        <value key="jscompilation">++resource++ftw.sliderblock/sliderblock-compiled.js</value>
+        <value key="merge_with">default</value>
```
Comment (jone) [inline ftw/sliderblock/upgrades/20191212135955_provide_precompiled_bundle/registry.xml:11]: `last_compilation`?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "ftw.sliderblock",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:4teamwork/ftw.solr.git",
+  "author": "4teamwork AG <info@4teamwork.ch>",
+  "license": "GPL2",
```
Comment (jone) [inline package.json:7]: ``` "private": true ``` is missing https://devdocs.4teamwork.ch/plone/javascript/ftw-resource-bundling-plone5/#bundling-with-yarn-and-friends-outside-plone

**Review by jone — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (Nachtalb) [reply package.json:7]: Done

File: ftw/sliderblock/profiles/default_plone5/registry.xml (hunk 2 above)
Comment (Nachtalb) [reply ftw/sliderblock/profiles/default_plone5/registry.xml:34]: Done

File: ftw/sliderblock/upgrades/20191212135955_provide_precompiled_bundle/registry.xml (hunk 3 above)
Comment (Nachtalb) [reply ftw/sliderblock/upgrades/20191212135955_provide_precompiled_bundle/registry.xml:11]: Done

File: ftw/sliderblock/browser/resources/sliderblock.js (hunk 1 above)
Comment (Nachtalb) [reply ftw/sliderblock/browser/resources/sliderblock.js:36]: I discussed it further with buchi and made changed the code accordingly

**Comment by Nachtalb** [comment]
For this change to work we need https://github.com/4teamwork/ftw.slider/pull/94. Should we wait for ftw.slider to be released and then change the required version in the setup.py?

File: ftw/sliderblock/browser/resources/sliderblock.js [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
-
+(function (root, factory) {
+    if (typeof define === 'function' && define.amd) {
+        // AMD. Register as an anonymous module.
+        define(['jquery', 'slider'], factory);
+    } else {
+        // Browser globals
+        factory(root.jQuery, root.Slider);
+    }
+}(typeof self !== 'undefined' ? self : this, function ($, Slider) {
```
Comment (jone) [inline ftw/sliderblock/browser/resources/sliderblock.js:9]: Is `typeof self !== 'undefined' ? self : this` something that we should do be default and that should be updated in the documentation at https://devdocs.4teamwork.ch/plone/javascript/ftw-resource-bundling-plone5/#transform-our-javascript-in-the-product ?

**Review by jone — APPROVED** [review]
> For this change to work we need [4teamwork/ftw.slider#94](https://github.com/4teamwork/ftw.slider/pull/94). Should we wait for ftw.slider to be released and then change the required version in the setup.py?

So we have a dependency. I agree to cut a release of `ftw.slider` and then put a `>=` version pinning into `setup.py`.

**Comment by jone** [comment]
I'll push a commit with a version pinning of `ftw.slider` and then merge and release.

**Pull request merged**